coordinator/src/main/scala/filodb.coordinator/client/IngestionOps.scala

```diff
@@ -1,38 +1,63 @@
 package filodb.coordinator.client
 
+import com.typesafe.scalalogging.slf4j.StrictLogging
 import scala.concurrent.duration._
 
 import filodb.core._
 import filodb.coordinator._
 
-trait IngestionOps extends ClientBase {
-  import NodeCoordinatorActor._
+trait IngestionOps extends ClientBase with StrictLogging {
+  import IngestionCommands._
 
   /**
    * Flushes the active memtable of the given dataset and version, no matter how much is in the memtable.
    * This should really only be done at the end of batch ingestions, to ensure that all data is finished
    * ingesting.  Calling this prematurely results in smaller chunks and less efficient storage.
+   * All NodeCoordinators will be asked to flush, and the count of nodes flushing will be returned.
    * @param dataset the DatasetRef of the dataset to flush
    * @param version the version of the dataset to flush
    * @throws ClientException
    */
-  def flush(dataset: DatasetRef, version: Int, timeout: FiniteDuration = 30.seconds): Unit = {
-    askCoordinator(Flush(dataset, version), timeout) {
-      case Flushed =>
-    }
+  def flush(dataset: DatasetRef, version: Int, timeout: FiniteDuration = 30.seconds): Int = {
+    askAllCoordinators(Flush(dataset, version), timeout) {
+      case Flushed => true
+      case FlushIgnored => false
+      case UnknownDataset => false
+    }.filter(x => x).length
   }
 
   def flushByName(datasetName: String,
                   database: Option[String] = None,
-                  version: Int = 0): Unit = flush(DatasetRef(datasetName, database), version)
+                  version: Int = 0): Int = flush(DatasetRef(datasetName, database), version)
 
   /**
-   * Retrieves the current ingestion stats for a particular dataset and version.
+   * Retrieves the current ingestion stats from all nodes for a particular dataset and version.
    */
   def ingestionStats(dataset: DatasetRef,
                      version: Int,
-                     timeout: FiniteDuration = 30.seconds): DatasetCoordinatorActor.Stats =
-    askCoordinator(GetIngestionStats(dataset, version), timeout) {
-      case stats: DatasetCoordinatorActor.Stats => stats
+                     timeout: FiniteDuration = 30.seconds): Seq[DatasetCoordinatorActor.Stats] =
+    askAllCoordinators(GetIngestionStats(dataset, version), timeout) {
+      case stats: DatasetCoordinatorActor.Stats => Some(stats)
+      case UnknownDataset => None
+    }.flatten
+
+  /**
+   * Repeatedly checks the ingestion stats and flushes until everything is flushed for a given dataset.
+   */
+  def flushCompletely(dataset: DatasetRef, version: Int, timeout: FiniteDuration = 30.seconds): Int = {
+    var nodesFlushed: Int = 0
+
+    def anyRowsLeft: Boolean = {
+      val stats = ingestionStats(dataset, version, timeout)
+      logger.debug(s"Stats from all ingestors: $stats")
+      stats.map(_.numRowsActive).filter(_ >= 0).sum > 0
```

Comment:
You can iterate just once vs twice with

       stats collect { case s if s.numRowsActive >= 0 => s }.sum > 0